packages/core/src/resources/MergeRequestApprovals.ts

```diff
@@ -143,25 +187,27 @@ export class MergeRequestApprovals<C extends boolean = false> extends BaseResour
       url = `projects/${pId}/approval_rules/${aId}`;
     }
 
-    return RequestHelper.put()(this, url, { name, approvalsRequired, ...options });
+    return RequestHelper.put<MergeRequestApprovalRuleSchema>()(this, url, {
+      name,
+      approvalsRequired,
+      ...options,
+    });
   }
 
-  editApprovals(
+  editApprovalsRequired(
```

Comment:
Id change this to editApprovalConfig for consistency

Reply:
Also, why remove the support for the project level edits? https://docs.gitlab.com/ee/api/merge_request_approvals.html#change-configuration

Reply:
Added `editProjectApprovalConfig` for the project level edits. The name may need to be changed but I don't think the MR and Project approval configs should be in the same method as they take very different parameters. Let me know what you think.